python/tidy/linting_report.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2013 The Servo Project Developers. See the COPYRIGHT
+# file at the top-level directory of this distribution.
+#
+# Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+# http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+# <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+# option. This file may not be copied, modified, or distributed
+# except according to those terms.
+
+
+from typing import List, Literal, TypedDict
+
+import colorama
+
+
+class RequiredAnnotation(TypedDict):
```

Comment:
I'm not sure if we need this base class. We always instantiate `OptionalAnnotation` and that is marked as partial (`total=False`) so none of the call sites can take advantage of the type checker to ensure all attributes are specified.